src/vellum/workflows/nodes/displayable/bases/inline_prompt_node/tests/test_inline_prompt_node.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+from typing import List
+
+from vellum import (
+    ChatMessagePromptBlock,
+    JinjaPromptBlock,
+    PlainTextPromptBlock,
+    PromptBlock,
+    RichTextPromptBlock,
+    VariablePromptBlock,
+)
+from vellum.workflows.errors import WorkflowErrorCode
+from vellum.workflows.exceptions import NodeException
+from vellum.workflows.nodes.displayable.bases.inline_prompt_node import BaseInlinePromptNode
+
+
+def test_extract_required_input_variables_with_various_block_types():
+    """Test that _extract_required_input_variables correctly identifies variables from different block types."""
+    test_blocks: List[PromptBlock] = [
+        VariablePromptBlock(input_variable="var1"),
+        ChatMessagePromptBlock(
+            chat_role="SYSTEM",
+            blocks=[
+                VariablePromptBlock(input_variable="var2"),
+            ],
+        ),
+        RichTextPromptBlock(
+            blocks=[
+                PlainTextPromptBlock(text="Some text"),
+                VariablePromptBlock(input_variable="var3"),
+            ],
+        ),
+        JinjaPromptBlock(template="Template without variables"),
+        ChatMessagePromptBlock(
+            chat_role="USER",
+            blocks=[
+                RichTextPromptBlock(
+                    blocks=[
+                        PlainTextPromptBlock(text="Nested text"),
+                        VariablePromptBlock(input_variable="var4"),
+                    ],
+                ),
+            ],
+        ),
+    ]
+
+    # GIVEN a BaseInlinePromptNode
+    class TestNode(BaseInlinePromptNode):
+        ml_model = "test-model"
+        blocks = test_blocks
+        prompt_inputs = {}
+
+    # WHEN the _extract_required_input_variables method is called
+    node = TestNode()
+    required_vars = node._extract_required_input_variables(test_blocks)
```

Comment:
[nit] in general, we should not need to be testing private methods, since it makes it harder to update implementation by requiring updates to these tests.

Instead, we should be able to call run and our assertion on the exception should give us the same information as the assertion below. Same feedback for all of the tests below